db/compaction_iterator.cc

```diff
@@ -187,6 +208,11 @@ void CompactionIterator::NextFromInput() {
       current_user_key_sequence_ = kMaxSequenceNumber;
       current_user_key_snapshot_ = 0;
 
+      if (all_versions_)
+        compaction_filter_->Callback(compaction_->level(), ikey_.user_key,
+                                     fromInternalValueType(ikey_.type), value_,
+                                     ikey_.sequence, true);
```

Comment:
Yes, I am having difficulty in moving away from this model. Event Listener/Compaction callback is a direct way of knowing what is going on under the hood. Using a polling based from approach from BlobDB to LSM seems to me wasteful and indirect.